python/lsst/drp/tasks/wcsfit.py

```diff
@@ -0,0 +1,1019 @@
+
+                wcsf.setObjects(extensionIndex, d, 'x', 'y', ['xCov', 'yCov', 'xyCov'])
+
+    def _addRefObjects(self, wcsf):
+        """Add reference sources to the wcsfit.WCSFit object.
+
+        Parameters
+        ----------
+        wcsf : `wcsfit.WCSFit`
+            WCS-fitting object
+        """
+        extensionIndex = self.extensionInfo[self.extensionInfo['extensionType'] == 'REFERENCE'].index[0]
+
+        if self.config.useProperMotion:
+            wcsf.setObjects(extensionIndex, self.refObjects, 'ra', 'dec', ['raCov', 'decCov', 'raDecCov'],
+                            pmDecKey='decPM', pmRaKey='raPM', parallaxKey='parallax', pmCovKey='fullCov',
+                            pmCov=self.refCovariance)
+        else:
+            wcsf.setObjects(extensionIndex, self.refObjects, 'ra', 'dec', ['raCov', 'decCov', 'raDecCov'])
+
+    def _makeAfwWcs(self, mapDict, centerRA, centerDec, doNormalizePixels=False, xScale=1, yScale=1):
+        """Make an `lsst.afw.geom.SkyWcs` from a dictionary of mappings
+
+        Parameters
+        ----------
+        mapDict : `dict`
+            Dictionary of mapping parameters
+        centerRA : `lsst.geom.Angle`
+            RA of the tangent point
+        centerDec : `lsst.geom.Angle`
+            Declination of the tangent point
+        doNormalizePixels : `bool`
+            Whether to normalize pixels so that range is [-1,1]
+        xScale : `float`
+            Factor by which to normalize x-dimension. Corresponds to width of
+            detector
+        yScale : `float`
+            Factor by which to normalize y-dimension. Corresponds to height of
+            detector
+
+        Returns
+        -------
+        outWCS : `lsst.afw.geom.SkyWcs`
+            WCS constructed from the input mappings
+        """
+        # Set up pixel frames
+        pixelFrame = astshim.Frame(2, "Domain=PIXELS")
+        normedPixelFrame = astshim.Frame(2, "Domain=NORMEDPIXELS")
+
+        if doNormalizePixels:
+            # Pixels will need to be rescaled before going into the mappings
+            normCoefficients = [-1.0, 2.0/xScale, 0,
+                                -1.0, 0, 2.0/yScale]
+            normMap = convertToAstPolyMapCoefficients(normCoefficients)
+        else:
+            normMap = astshim.UnitMap(2)
+
+        # All of the detectors for one visit map to the same tangent plane
+        tangentPoint = lsst.geom.SpherePoint(centerRA, centerDec)
+        cdMatrix = afwgeom.makeCdMatrix(1.0 * lsst.geom.degrees, 0 * lsst.geom.degrees, True)
+        iwcToSkyWcs = afwgeom.makeSkyWcs(lsst.geom.Point2D(0, 0), tangentPoint, cdMatrix)
+        iwcToSkyMap = iwcToSkyWcs.getFrameDict().getMapping("PIXELS", "SKY")
+        skyFrame = iwcToSkyWcs.getFrameDict().getFrame("SKY")
+
+        frameDict = astshim.FrameDict(pixelFrame)
+        frameDict.addFrame("PIXELS", normMap, normedPixelFrame)
+
+        currentFrameName = "NORMEDPIXELS"
+
+        # Assume python 3.7+ so dict is ordered:
```

Comment:
We're on py3.10 now, so this comment isn't necessary.

Reply:
I rewrote this to explain why the order matters.